pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/PythonExtension.groovy

```diff
@@ -141,6 +143,8 @@ class PythonExtension {
         if (pythonEnvironment.containsKey('PYGRADLE_PROJECT_VERSION')) {
             throw new GradleException("Cannot proceed with `PYGRADLE_PROJECT_VERSION` set in environment!")
         }
+
+        project.getExtensions().add("zipapp", zipapp)
```

Comment:
I don't think this is going to work.  As a backward compatibility hack, it needs to be installed on the project, because for example, if we try to add this in the `PexExtension` constructor, the `PythonExtension` isn't registered yet.  So you either get an exception there, or you get a null pointer exception later (e.g. in `PythonWebApplicationPluginTest`).

Reply:
I'm going to resolve this conversation, but if you have an idea that we can make work, I'll give it a try.

Reply:
Resolved offline